horace_core/sqw/PixelData/@PixelTmpFileHandler/PixelTmpFileHandler.m

```diff
@@ -51,6 +51,48 @@
         raw_pix = fread(file_id, page_shape, obj.FILE_DATA_FORMAT_);
     end
 
+    function raw_pix = load_pixels_at_indices(obj, page_number, indices, ncols)
+        if nargin == 2
+            ncols = 1;
+        end
+
+        NUM_BYTES_IN_FLOAT = 4;
+        PIXEL_SIZE = NUM_BYTES_IN_FLOAT*ncols;  % bytes
+
+        tmp_file_path = obj.generate_tmp_pix_file_path_(page_number);
+        [file_id, err_msg] = fopen(tmp_file_path, 'rb');
+        if file_id < 0
+            error('PIXELTMPFILEHANDLER:load_page', ...
+                  'Could not open ''%s'' for reading:\n%s', tmp_file_path, ...
+                  err_msg);
+        end
+        clean_up = onCleanup(@() fclose(file_id));
+
+        indices_monotonic = issorted(indices, 'strictascend');
+        if ~indices_monotonic
+            [indices, ~, idx_map] = unique(indices);
+        end
+
+        [read_sizes, seek_sizes] = get_read_and_seek_sizes(indices);
+
+        raw_pix = zeros(ncols, numel(indices));
+
+        num_pix_read = 0;
+        for block_num = 1:numel(read_sizes)
+            fseek(file_id, seek_sizes(block_num)*PIXEL_SIZE, 'cof');
+
+            out_pix_start = num_pix_read + 1;
+            out_pix_end = out_pix_start + read_sizes(block_num) - 1;
+            read_pix = fread(file_id, [ncols, read_sizes(block_num)], obj.FILE_DATA_FORMAT_);
```

Comment:
there's no error handling with this `fread` or `fseek` - can you reuse `do_fread` and `do_fseek` defined (currently) as part of `sqw_binfile_common`?
These are generic file reading helpers so could reasonably move into the utilities folder